### ISSUE TITLE
Creating route alias using post processed fixture data (Blob) is not returning that data

### ISSUE THREAD

**Issue by saschwarz** [cypress-io/cypress#1956]
### Current behavior:

My Angular 5 app makes an HTTP GET request to a remote server which returns an image as a Blob. My existing Cypress specs using the remote server correctly verify the response data is received.

I'd like to stub out the call to the remote server and wrote this spec that loads a JPEG file from disk using a fixture (which returns a dataURI) and then I convert that to a Blob and supply it to `cy.route` (using the same route minimatch expression that is matched in my non-stubbed request).

in the cypress command log the XHR Stub is being hit and it is returning a Blob of size 2 of type application/json instead of my actual Blob. The `cy.wait` for the alias also prints Blob size2 application/json to the console.

Here's my spec:

```js
// spec
describe('Test using fixture to supply image as a Blob', function () {
  before(() => {
    cy.server();
    cy.fixture('images/map.jpeg').then(dataURI => {
      console.log(dataURI);
       return Cypress.Blob.base64StringToBlob(dataURI, "image/jpeg").then((blob) => {
        console.log(blob);
        return cy.route('https://images.mydomain.com/**', blob).as('imageXHR');
      })
    });
  })
  it('test that causes app to call '@imageXHR'), function() {
    ...
  })
});
```
The `console.log` statements print what you'd expect - the first is a long data URI and the second is 
a `Blob {size: 41191, type: "image/jpeg"}`. 

![screen shot 2018-06-14 at 9 03 38 pm](https://user-images.githubusercontent.com/184171/41447102-25381da8-7019-11e8-872f-b9f3d914069c.png)

I'm not sure what syntax I have wrong or if there is a bug.

### Desired behavior:

The post processed fixture data is supplied to the XHR request.

### Steps to reproduce:

I'm happy to work on a reproduction if it would help!

### Versions

Cypress 3.0.1, OSX

**Comment by brian-mann** [comment]
Have you tried just passing the fixture reference directly as opposed to trying to convert it to a `Blob` ?

Pinging @bahmutov as this has to do with the network layer which we're current in the middel of a rewrite. Good scenario to account for if its not possible to get working with our current model.

Yes @saschwarz please put together a tiny reproducible repo so we have the minimum to test and understand your situation.

**Comment by saschwarz** [comment]
@brian-mann I've created a test: https://github.com/saschwarz/cypress-test-tiny
It has the three variations I've tried for stubbing my jpeg file as a Blob with mime type text/binary.

Let me know if there is anything else I can do.

**Comment by matt328** [comment]
I have the exact same problem, Cypress@3.1.1, no matter what I try, nothing other than using `'fx:data.json'` as the last arg to `cy.route()` works.

Especially frustrating is this snipped logs my data, but the route always returns no response:

```typescript
cy.fixture('data.json').as('data');
cy.get('@data').then(d => {
   // Logs data to the console
   console.log({ d });
   cy.route('GET', '/some/url/**', d);
});

```

**Comment by amcgee** [comment]
Just stumbled on this issue, here's a workaround.

@saschwarz Thanks for the test repo!  This is definitely a bug, but as a workaround I was able to modify your `variant1` as follows and the test now passes, yielding a fixture-sourced image blob:

```js
cy.fixture('images/map.jpeg').then(dataURI => {
    return Cypress.Blob.base64StringToBlob(dataURI, "image/jpeg").then(
        (blob) => {
            expect(blob.size).to.be.equal(41191);
            expect(blob.type).to.equal('image/jpeg');
            return cy.route({
              url: 'https://images.agilitycoursemaster.com/**',
              method: 'GET',
              response: '',
              onResponse: xhr => {
                xhr.response.body = blob;
              }
            }).as('imageXHR');
        })
});
```

The important bit is this:
```js
onResponse: xhr => {
  xhr.response.body = blob;
}
```

It's a hack, but it should unblock you.

@brian-mann *Cypress is great, thanks for making it*

**Comment by jennifer-shehane** [comment]
Issue is reproducible in https://github.com/saschwarz/cypress-test-tiny repo in version 3.2.0.

<img width="460" alt="Screen Shot 2019-04-25 at 2 24 21 PM" src="https://user-images.githubusercontent.com/1271364/56718969-d6866300-6765-11e9-9c5a-01ab54631d55.png">
 
<img width="459" alt="Screen Shot 2019-04-25 at 2 24 37 PM" src="https://user-images.githubusercontent.com/1271364/56718994-e30abb80-6765-11e9-9f5a-6fda87feabe3.png">

**Comment by peterpeterparker** [comment]
@jennifer-shehane may I  ask  if this bug is still on the roadmap or has it been solved?

I tried above workaround of @amcgee but I am unfortunately still  unable to mock an Http Get request with a blob result, at least in an Angular project.

If interested I can provide the following dummy sample repo:

git clone https://github.com/peterpeterparker/cypress-mock-blob
cd cypress-mock-blob
npm i
npm run cypress

select test "mock-get-blob.js" to run and observe the console. instead of a mocked Http answer, I get a 0 length text/plain answer

Thx in advance for the feedback

<img width="1312" alt="Capture d’écran 2020-07-01 à 15 59 49" src="https://user-images.githubusercontent.com/16886711/86252819-f7d9af00-bbb3-11ea-940c-eb22753733b1.png">

**Comment by lucax88x** [comment]
I'm having the same identical issue of @peterpeterparker

**Comment by Airatiki** [comment]
On 5.3.0 I have the same identical issue of @peterpeterparker. But I rewrite to 
`return cy.route({
            url,
            method: 'GET',
            response: 'fx:image.png,binary', // where image.png my file in fixtures folder
            headers: {
                'content-type': 'application/octet-stream',
            },
        })`
and now it's working

**Comment by ClaudiaStefan** [comment]
@Airatiki your suggestion worked for me as well. Thanks.

**Comment by mpavel** [comment]
The example above by @Airatiki should be in the documentation 😅  👏

**Comment by peterpeterparker** [comment]
@Airatiki have you got maybe a all block of test to share? I am back on this issue after a while, it would help me to understand how you plugged the route. thx in advance

**Comment by peterpeterparker** [comment]
ah yeah gotcha, thx  @Airatiki 

The modified version of the project I linked  above with the solution would be:

```
beforeEach(() => {
        cy.server();

        cy.fixture('a.pdf', 'base64').then(dataURI => {
            return cy
                .route({
                    url: /\/myroute\/.*/g,
                    method: 'GET',
                    response: 'fx:a.pdf,binary',
                    headers: {
                        'content-type': 'application/octet-stream'
                    }
                })
                .then(() => {
                    return cy.visit(`http://localhost:4200/`);
                });
        });
    });
```

**Comment by peterpeterparker** [comment]
It even seems that this is solved with Cypress v6 and the use of [intercept](https://docs.cypress.io/api/commands/intercept.html#Arguments).

Following work as expected  without workaround.

```
cy.intercept('/myroute', { fixture: 'a.pdf' }).then(() => {
        return cy.visit(
            'http://localhost:4200/'
        );
    });
````

**Comment by emilyrohrbough** [comment]
In the upcoming v12.0.0 release, the deprecated `cy.route()` will be removed. This has been replaced with [`cy.intercept()`](https://docs.cypress.io/api/commands/intercept). Closing as won't fix.
